### PR TITLE
feat(track-state): Log on add/remove/mute/owner.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

File: conference.js [hunk 1]
```diff
@@ -845,7 +845,7 @@ export default {
             // This will only modify base/media.audio.muted which is then synced
             // up with the track at the end of local tracks initialization.
             muteLocalAudio(mute);
-            this.setAudioMuteStatus(mute);
+            this.updateAudioIconEnabled();
```
Comment (jallamsetty1) [inline conference.js:848]: Do we even need to call it here? What is the purpose of checking here if the audio icon needs to be enabled or disabled.

File: conference.js [hunk 2]
```diff
@@ -1394,7 +1394,7 @@ export default {
                 APP.store.dispatch(
                 replaceLocalTrack(oldTrack, newTrack, room))
                     .then(() => {
-                        this.setAudioMuteStatus(this.isLocalAudioMuted());
+                        this.updateAudioIconEnabled();
```
Comment (jallamsetty1) [inline conference.js:1397]: Ditto

File: react/features/base/tracks/middleware.web.ts [hunk 3]
```diff
@@ -113,9 +112,7 @@ MiddlewareRegistry.register(store => next => action => {
                 APP.UI.setVideoMuted(participantID);
             }
         } else if (jitsiTrack.isLocal()) {
-            APP.conference.setAudioMuteStatus(muted);
-        } else {
-            APP.UI.setAudioMuted(participantID, muted);
+            APP.conference.updateAudioIconEnabled();
```
Comment (jallamsetty1) [inline react/features/base/tracks/middleware.web.ts:150]: Same question here again. If we already know that a local audio track exists, why do we need to check if the mic icon needs to be enabled or disabled?

File: react/features/base/tracks/middleware.web.ts [hunk 4]
```diff
@@ -62,7 +71,32 @@ MiddlewareRegistry.register(store => next => action => {
 
     case TRACK_REMOVED: {
         _removeNoDataFromSourceNotification(store, action.track);
-        break;
+
+        const result = next(action);
```
Comment (jallamsetty1) [inline react/features/base/tracks/middleware.web.ts:75]: Do we need to log it here since with ssrc-rewriting, tracks will never be removed.

Comment (hristoterezov) [reply react/features/base/tracks/middleware.web.ts:75]: It could be useful for the case where the ssrc rewriting is not enabled.

**Comment by hristoterezov** [comment]
About your questions related to the first commit:
Currently I don't have a good answer. I just noticed that we are calling a function from UI that calls a function from conference.js and works only for local tracks. So basically my idea was to simplify the code by leaving only the function in conference.js. Therefore the current code execution will do exactly what the previous version was doing. 

Will check on the specific places that you pointed out if we actually we need to call the function although IMHO this is totally different story.

File: react/features/base/tracks/middleware.web.ts (hunk 4 above)
Comment (jallamsetty1) [reply react/features/base/tracks/middleware.web.ts:75]: Even without ssrc rewriting, remote tracks are removed only when participants leave. Do you think track states will be helpful in that case. I am just worried about all the unnecessary logging.

**Comment by hristoterezov** [comment]
It seems that we try to update it whenever we do changes to the local tracks or devices. On some places it may be redundant but I think the whole thing needs to be refactored. So lets leave it for another PR.

**Review by jallamsetty1 — APPROVED** [review]

**Pull request merged**